GLOSSARY.md

```diff
@@ -0,0 +1,284 @@
+will be expanded in the future. **Please avoid changing existing headings as
+users may be deep-linking to the generated anchors.**
+
+## Table of Contents
+
+* [Error Symposium](#error-symposium)
+  * [Operational Error](#operational-error)
+  * [Programmer Error](#programmer-error)
+  * [Expected versus Unexpected](#expected-versus-unexpected)
+* [Post Mortem Analysis](#post-mortem-analysis)
+  * [Abort](#abort)
+  * [Core / Core dump / Core file](#core--core-dump--core-file)
+  * [Stack Frame](#stack-frame)
+  * [Unwinding the Stack / Top of Stack](#unwinding-the-stack--top-of-stack)
+* [Promise](#promise)
+  * [Executor / Resolver](#executor--resolver)
+  * [Handler](#handler)
+  * [Settling](#settling)
+    * [Rejection / reject](#rejection--reject)
+    * [Resolving / resolve](#resolving--resolve)
+  * [Synchronous Rejection](#synchronous-rejection)
+  * [Unhandled Rejection / Rejection Handled](#unhandled-rejection--rejection-handled)
+  * [Microtask Queue](#microtask-queue)
+* [Links](#links)
+
+## Error Symposium
+
+An ongoing series of facilitated discussions led by **@groundwater**, originally
+conceived at the August 2015 Node Collaborator Summit at Mozilla SF. Their goal
+is to produce recommendations on "best practices" for error handling in Node.
+
+Notably, the conversation has been informed by the [foundational work][joyent-errors]
+written by **@davepacheco**. This document was paraphrased by **@chrisdickinson** as
+the part of the Node.js [error documentation][node-errors].
+
+[The results of their latest meeting are here][symposium-findings].
+
+### Operational Error
+
+An error that represents a known state of the program. To paraphrase the
+[joyent error guide][joyent-errors]:
+
+> Operational errors represent run-time problems experienced by correctly-written
+> programs. These are not bugs in the program. In fact, these are usually
+> problems with something else: the system itself (e.g., out of memory or too
+> many open files), the system's configuration (e.g., no route to a remote host),
+> the network (e.g., socket hang-up), or a remote service (e.g., a 500 error,
+> failure to connect, or the like).
+
+Notably, operational errors that are not accounted for by the programmer are
+considered unexpected errors.
+
+### Programmer Error
+
+An error that represents a mistake on the programmer's behalf. To paraphrase the
+[joyent error guide][joyent-errors]:
+
+> Programmer errors are bugs in the program. These are things that can always
+> be avoided by changing the code. They can never be handled properly (since by
+> definition the code in question is broken).
+
+By definition, these errors are unexpected.
+
+### Expected versus Unexpected
+
+Errors may be "expected" — anticipated by the program author at time of writing —
+or "unexpected" — unforeseen by the author.
+
+Programmer errors are _always_ unexpected (or else the programmer would not
```

Comment:
I don't wish to change or affect the definitions of programmer error/operational error as they are defined by error symposioum etc.

I wish to define separate error categories: "unexpected" and "expected" errors with the clear distinction that these will be consumer decided (thus they don't have analogy to the error symposium definitions which are producer decided). If these "unexpected" and "expected" are also already taken by error symposium then we need to call them something else.